src/main/java/org/manifold/compiler/front/FrontendBuildException.java

```diff
@@ -0,0 +1,12 @@
+package org.manifold.compiler.front;
+
+/**
+ * Use this for errors that should be reported directly to the user.
+ * Non-internal errors should be translated to a FrontendBuildException somewhere in the pipeline.
+ */
+public class FrontendBuildException extends Error {
```

Comment:
Hey, been a while but I'm still getting these and this one caught my eye. You probably want to extend Exception here instead of Error, which is reserved for serious things that shouldn't be caught, eg. OutOfMemory and StackOverflow.